### PR TITLE
Head Recursion

### PR DESCRIPTION
**Pull request by i-am-aisha**
## Related Issue
#1330
## Proposed Changes
Create a HeadRecursion.cpp code file with mentioning proper comments and code of head recursion in super easy and understandable way.
updated readme accordingly to the change in folder types of recursion.

## Checklist

- [x] ✅ My code follows the code style of this project.
- [x] 📝 My change requires a change to the README.
- [x] 🎀 I have updated the README accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

## Output Screenshots
code images
Screenshot (17)

Input image
Screenshot (15)
output image
Screenshot (16)

### CODE REVIEW

File: R-Recursion/Types of Recursion/README.md [hunk 1]
```diff
@@ -54,3 +58,32 @@ we call the function fib(4) which generates two more calls fib(3) and fib(2) , f
...
+ If a recursive function calling itself and that recursive call is the first statement in the function then it’s known as Head Recursion. There’s no statement, no operation before the call. The function doesn’t have to process or perform any operation at the time of calling and all operations are done at returning time.
+
+Let’s understand the example by tracing tree of recursive function. That is how the calls are made and how the outputs are produced.
+![image](https://user-images.githubusercontent.com/100334178/166974790-912ef94f-0e4a-4730-a374-25cf4dd146da.png)
+
+*  If there is something before the recursive call then it is not a head recursion. If something is there before the function call, it is just a recursion. 
+
+
+## Algorithm
+we pass n=5 in the function fun, each time call for function with value n-1 without printing it at first. As a result, fun(1) will be the first to execute completely and 1 is printed first. The function calls terminate when n=0, this is the Base Case of our Recursive function. We can see the call is not the last step in the function. The program above demonstrates Given a Number N Print 1 to N without Loop.
```
Comment (Mrjoy832) [inline R-Recursion/Types of Recursion/README.md:73]: write algo using some points

**Review by Mrjoy832 — CHANGES_REQUESTED** [review]

File: R-Recursion/Types of Recursion/README.md (hunk 1 above)
Comment (i-am-aisha) [reply R-Recursion/Types of Recursion/README.md]: updated 👍

**Comment by i-am-aisha** [comment]
@Mrjoy832 I have updated the changes please have a look on it.

File: R-Recursion/Types of Recursion/README.md [hunk 2]
```diff
@@ -10,8 +10,26 @@
       - [Disadvantages](#disadvantages)
 
 
+
+# Different types of the recursion
+* Direct Recursion.
+* Indirect Recursion.
+* Tail Recursion.
+- [Head Recursion](#head-recursion)
+  - [Algorithm](#head-recursion)
```
Comment (Mrjoy832) [inline R-Recursion/Types of Recursion/README.md:19]: image
At the top Table of Content is not properly done as there Only Tree Recursion mentioned..kindly remove and make it better

**Review by Mrjoy832 — CHANGES_REQUESTED** [review]

File: R-Recursion/Types of Recursion/README.md (hunk 2 above)
Comment (i-am-aisha) [reply R-Recursion/Types of Recursion/README.md]: Screenshot (26)
 @Mrjoy832  Hope now its fine.

**Comment by Mrjoy832** [comment]
LGTM🌟

**Review by Lakhankumawat — APPROVED** [review]
LGTM ✨🎉

**Pull request merged**